### ISSUE TITLE
Assertion in QuickInfo for Async.StartImmediateSafe

### ISSUE THREAD

**Issue by vasily-kirichenko** [Microsoft/visualfsharp#2942]
1. Open VFPT solution
2. Hover over the following symbol:

![image](https://cloud.githubusercontent.com/assets/873919/25535288/3135c76e-2c3f-11e7-967b-042868bbf926.png)

3. Result:
```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
Unexpected exception seen in language service

System.IndexOutOfRangeException: Index was outside the bounds of the array.

   at System.String.get_Chars(Int32 index)

   at Microsoft.VisualStudio.FSharp.Editor.Tokenizer.tokens@461-1.Invoke(Int32 index, FSharpTokenInfo token)

   at Microsoft.VisualStudio.FSharp.Editor.Extensions.List.foldi[State,T](FSharpFunc`2 folder, State state, FSharpList`1 xs)

   at Microsoft.VisualStudio.FSharp.Editor.Tokenizer.getSymbolFromTokens(String fileName, FSharpList`1 tokens, LinePosition linePos, String lineStr, SymbolLookupKind lookupKind, Boolean wholeActivePatterns)

   at Microsoft.VisualStudio.FSharp.Editor.Tokenizer.getSymbolAtPosition(DocumentId documentKey, SourceText sourceText, Int32 position, String fileName, FSharpList`1 defines, SymbolLookupKind lookupKind, Boolean wholeActivePatterns)

   at Microsoft.VisualStudio.FSharp.LanguageService.Assert.Exception(Exception e)

   at Microsoft.VisualStudio.FSharp.Editor.Tokenizer.getSymbolAtPosition(DocumentId documentKey, SourceText sourceText, Int32 position, String fileName, FSharpList`1 defines, SymbolLookupKind lookupKind, Boolean wholeActivePatterns)

   at Microsoft.VisualStudio.FSharp.Editor.FSharpQuickInfo.getTooltipInfo@91-2.Invoke(FSharpProjectOptions _arg2)

   at Microsoft.VisualStudio.FSharp.Editor.Pervasive.Bind@140-4.Invoke(Unit unitVar)

   at Microsoft.FSharp.Control.AsyncBuilderImpl.callA@839.Invoke(AsyncParams`1 args)

   at <StartupCode$FSharp-Core>.$Control.loop@426-38.Invoke(FSharpFunc`2 action)

   at Microsoft.FSharp.Control.Trampoline.ExecuteAction(FSharpFunc`2 firstAction)

   at Microsoft.FSharp.Control.TrampolineHolder.Protect(FSharpFunc`2 firstAction)

   at Microsoft.FSharp.Control.TaskHelpers.continuation@1550.Invoke(Task`1 completedTask)

   at Microsoft.FSharp.Control.TaskHelpers.continueWith@1560.Invoke(Task`1 obj)

   at System.Threading.Tasks.ContinuationTaskFromResultTask`1.InnerInvoke()

   at System.Threading.Tasks.Task.Execute()

   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)

   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)

   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)

   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()

   at System.Threading.ThreadPoolWorkQueue.Dispatch()

   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()


---------------------------
Abort   Retry   Ignore   
---------------------------
```

**Comment by dsyme** [comment]
Yeah I saw this too.  I presume [this](https://github.com/Microsoft/visualfsharp/pull/2929/commits/cdefd7bde38b9257e9c7cf7715b58d5912de9b70) fixes it, included in #2929

**Comment by vasily-kirichenko** [comment]
I confirm it's fixed in #2929 

![image](https://cloud.githubusercontent.com/assets/873919/25556881/0e6849da-2d0e-11e7-9ff1-7f757c04f5cc.png)

**Comment by dsyme** [comment]
@vasily-kirichenko Thanks for checking this
